service/Microsoft.Azure.Devices/X509ThumbprintExtensions.cs

```diff
@@ -6,65 +6,53 @@ namespace Microsoft.Azure.Devices
     using System;
     using System.Text.RegularExpressions;
     using Microsoft.Azure.Devices.Common;
-    using Newtonsoft.Json;
 
     /// <summary>
     /// X509 client certificate thumbprints of the device
     /// </summary>
-    public sealed class X509Thumbprint
+    public static class X509ThumbprintExtensions
     {
         static readonly Regex ThumbprintRegex = new Regex(@"^([A-Fa-f0-9]{2}){20}$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        /// <summary>
-        /// X509 client certificate primary thumbprint
-        /// </summary>
-        [JsonProperty(PropertyName = "primaryThumbprint")]
-        public string PrimaryThumbprint { get; set; }
-
-        /// <summary>
-        /// X509 client certificate secondary thumbprint
-        /// </summary>
-        [JsonProperty(PropertyName = "secondaryThumbprint")]
-        public string SecondaryThumbprint { get; set; }
 
         /// <summary>
         /// Checks if contents are valid
         /// </summary>
         /// <param name="throwArgumentException"></param>
         /// <returns>bool</returns>
-        public bool IsValid(bool throwArgumentException)
+        public static bool IsValid(this X509Thumbprint x509Thumbprint, bool throwArgumentException)
```

Comment:
This is a binary breaking change. Not sure if it can be solved via TypeForwarding.